backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/optimizations/Devirtualization.kt

```diff
@@ -97,7 +97,9 @@ internal object Devirtualization {
                 moduleDFG.symbolTable.functionMap.values.filter { it.explicitlyExported } +
                 externalModulesDFG.functionDFGs.keys.filter { it.explicitlyExported }
 
-        // Conservatively assume each associated object could be instantiated.
+        // Conservatively assume each associated object could be called.
+        // Note: for constructors there is additional parameter (<this>) and its type will be added
```

Comment:
Looks fragile.
Please add a comment about this expectation to the code that adds final types to instantiating classes.